Mathlib/Algebra/Polynomial/AlgebraMap.lean

```diff
@@ -294,12 +294,32 @@ def algEquivOfCompEqX (p q : R[X]) (hpq : p.comp q = X) (hqp : q.comp p = X) : R
   refine AlgEquiv.ofAlgHom (aeval p) (aeval q) ?_ ?_ <;>
     exact AlgHom.ext fun _ ↦ by simp [← comp_eq_aeval, comp_assoc, hpq, hqp]
 
+@[simp]
+theorem algEquivOfCompEqX_eq_iff (p q p' q' : R[X])
```

Comment:
Perhaps in all of those lemmas, the non-Prop arguments should be implicit. Will let a maintainer decide